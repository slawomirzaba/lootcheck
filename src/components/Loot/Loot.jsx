import React from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../../reducers/actions/bitcoin';

export class Loot extends React.Component {

    componentDidMount() {
        this.props.fetchBitcoin();
    }

    convertBitcoinBalance = () => {
        if (!this.props.bitcoin) return '';
        
        const balance = this.props.balance;
        const bitcoinBalance = parseFloat(this.props.bitcoin.bpi.USD.rate.replace(',', ''));

        return balance / bitcoinBalance;
    }

    render() {
        return (
            <h3>Balance Bitcoin: {this.convertBitcoinBalance()}</h3>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        balance: state.balance.balance,
        bitcoin: state.bitcoin.bitcoin
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchBitcoin: () => dispatch(fetchBitcoin())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Loot);