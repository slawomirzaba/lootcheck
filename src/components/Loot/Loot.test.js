import React from 'react';
import { shallow, mount } from 'enzyme';
import { Loot } from './Loot';

describe('Loot', () => {
    const rate = '10';
    const balance = 10000;
    const props = { balance, bitcoin: null, fetchBitcoin: () => { } };
    let loot = shallow(<Loot {...props} />);

    it('should correct render', () => {
        expect(Loot).toMatchSnapshot();
    });

    describe('when component did mount', () => {
        const mockFetch = jest.fn();
        props.fetchBitcoin = mockFetch;

        beforeEach(() => {
            loot = mount(<Loot {...props} />)
        });

        it('should call fetch bitcoin', () => {
            expect(mockFetch).toHaveBeenCalled();
        })
    });

    describe('when data have been already fetched', () => {
        props.bitcoin = { bpi: { USD: { rate } } }
        beforeEach(() => {
            loot = mount(<Loot {...props} />)
        });

        it('should display correct bitcoin balance', () => {
            expect(loot.find('h3').text()).toEqual('Balance Bitcoin: 1000');
        })
    });
});