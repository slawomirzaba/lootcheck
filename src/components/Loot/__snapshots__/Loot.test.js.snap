// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot should correct render 1`] = `[Function]`;
